src/lib/connectors/cdp/cdp-launcher.ts

```diff
@@ -31,7 +31,9 @@ export class CDPLauncher extends Launcher {
 
         chromeFlags.push('--no-default-browser-check');
         if (isCI) {
-            chromeFlags.push('--headless', '--disable-gpu');
+            // Running in `headless` mode is causing error in tests for `set-cookie` header.
+            // So the `--headless` flag was removed.
```

Comment:
And the `--headless` mode only ignores the `set-cookie` header, once I switched the header name to something else in the test,  like `strict-transport-security`,  the configured header is recognized in cdp again :(.